src/dsp/rx_fft.cpp

```diff
@@ -359,16 +359,9 @@ void rx_fft_f::do_fft(unsigned int size)
     gr_complex *dst = d_fft->get_inbuf();
     float * p = (float *)d_reader->read_pointer();
     /* apply window, and convert to complex */
+    volk_32f_x2_interleave_32fc(dst, p, p, size);
```

Comment:
Shouldn't the `q` buffer be zero here? Setting it to `p` increases the power of the signal by 3 dB, which makes the audio FFT inaccurate.

Reply:
Thanks.
Yes. This in inaccurate.
Keeping a buffer full of zeroes does not sound as a good idea.
I think, we can compensate for the error by dividing the window by 2, maybe with `volk_32f_s32f_multiply_32f` here https://github.com/gqrx-sdr/gqrx/blob/801aa9e43e83e23f16346a9d5cabee4b772e317a/src/dsp/rx_fft.cpp#L426
It would be called once as there is no way to change audio FFT parameters.
And when we'll switch to complex input to this FFT to show RAW IQ demodulator output correctly, we'll have to disable this scaling for this specific demodulator type.

Reply:
That sounds complicated, and I think someone reading the code might find it confusing that `rx_fft_f` computes the FFT of the signal multiplied by 1+i.

Maybe for now it would be simplest to remove the VOLK optimization from `rx_fft_f`. The audio rate is much lower than the input sample rate, so it should not be a performance bottleneck.

Reply:
OK.
I've reverted 'empty window' branch to trivial for-loop and rebased to current master.

Reply:
Thanks. The other branch still looks broken though?

Reply:
Sorry. I'm wrong. Wait a minute. I'll push correct commit.

Reply:
I think it would be best to limit this to `rx_fft_c` and `rx_meter_c` for now, since the calculations there are unchanged and those are the high-sample rate blocks.

Reply:
Ready.
This should be correct. I'll test on my Intel Atom N270 system to see if there would be any improvement from VOLK optimization of `rx_fft_f`.

Reply:
No noticeable difference even on Intel Atom N270. Let's keep it untouched and add a special case for handling complex output from 'RAW IQ' demodulator later.